pkg/agent/cniserver/ipam/antrea_ipam.go

```diff
@@ -232,3 +232,11 @@ func init() {
 		klog.Errorf("Failed to register IPAM plugin on type %s", ipamHostLocal)
 	}
 }
+
+func parseVLANID(vlanString string) *uint16 {
```

Comment:
This is what I do not like: use inefficient types and convert at all the consumers.

Reply:
Currently we have only 1 consumer (the line where calls parseVLANID) and 3 producers (AllocateNext/AllocateIP/AllocateReservedOrNext) for this data.
If we use an internal struct to carry this, we still need to call `parseVLANID` within those producers.

Reply:
I know. But I meant we should do the conversion at the earliest point for decoupling. Func like parseVLANID() should not be in the consumer side.